nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/heartbeat/AbstractHeartbeatMonitor.java

```diff
@@ -296,6 +305,16 @@ private void processHeartbeat(final NodeHeartbeat heartbeat) {
         clusterCoordinator.validateHeartbeat(heartbeat);
     }
 
+    private boolean isNodeApiReachable(final NodeIdentifier nodeIdentifier) {
+        try {
+            final InetAddress nodeApiAddress = InetAddress.getByName(nodeIdentifier.getApiAddress());
+            return nodeApiAddress.isReachable((int) NODE_API_TIMEOUT.toMillis());
```

Comment:
The [isReachable](https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#isReachable-int-) method can behave differently depending on the platform, sometimes using ICMP, and sometimes using TCP port 7. For a more reliable check, seems worth considering making a socket connection to the node port, which would ensure basic TCP connectivity.